// card/src/card-lit.js
import { LitElement, html, css } from 'lit-element';

import { generalStyles } from '../css/general-styles';
import { cardStyles } from '../css/card-styles';


export class CardComponent extends LitElement {

  static get styles(){
    return[
      generalStyles,
      cardStyles
    ]
  }

  static get properties(){
    return{
      title: {type:String},
      text: {type: String}
    }
  }

  constructor(){
    super()
    this.title = "Rick Sánchez";
    this.text = "Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only five centuries, but also the leap into electronic typesetting, remaining essentially unchanged. It was popularised in the 1960s with the release of Letraset sheets containing Lorem Ipsum passages, and more recently with desktop publishing software like Aldus PageMaker including versions of Lorem Ipsum.";
  }

  render() {
    return html`
      <div class="grid-column-4">
        <card-album class="card-album">
        <img src="img/rick_morty.png">
          <p class="title">${this.title}</p>
        </card-album>

        <card-album class="card-album">
        <img src="img/rick_morty.png">
          <p class="title">${this.title}</p>
        </card-album>

        <card-album class="card-album">
          <img src="img/rick_morty.png">
          <p class="title">${this.title}</p>
        </card-album>

        <card-album class="card-album">
        <img src="img/rick_morty.png">
          <p class="title">${this.title}</p>
        </card-album>
      </div>
      <br><br>
      <div class="grid-column-1">
        <card class="card">
          <p>${this.text}</p>
        </card>
      </div>
    `;
  }
}

customElements.define('cards-lit', CardComponent);
