// input/src/input-lit.js
import { LitElement, html, css } from 'lit-element';
import { inputStyles } from '../css/input-styles';

export class inputComponent extends LitElement {

  static get styles(){
    return[
      inputStyles
    ]
  }

  static get properties(){
    return{
      msgvalue: {type: String},
      placeHolder: {type: String},
      blankText:{type: String},
      maxLengthText: {type: Number},
      minLengthText: {type: Number},
      inputType: {type: String},
      disabled: {type: Boolean}
    }
  }

  constructor(){
    super()
    this.msgvalue = '';
    this.inputType = 'text';
    this.placeHolder = 'Test';
    this.blankText = "Campo obligatorio";
    this.minLengthText = 2;
    this.maxLengthText = 5;
    this.disabled = true;
  }

  render() {
    return html`
      <div-input>
        <label class="required">Input general</label>
        <div class="space"></div>
        <input class="colorInput"
        .value="${this.msgvalue}"
        .type="${this.inputType}"
        .placeholder="${this.placeHolder}"
        @change="${this.readInput}"
        @keyup="${this.requiredInput}"
        >
        <div class="space"></div>
        <span class="error-msg" id="error"></span>
      </div-input>

      <div-input>
      <label class="required">Input general</label>
      <div class="space"></div>
      <input class="colorInput"
      .value="${this.msgvalue}"
      .type="${this.inputType}"
      .placeholder="${this.placeHolder}"
      @change="${this.readInput}"
      @keyup="${this.requiredInput}"
      ?disabled=${this.disabled}      
      >
      <div class="space"></div>
      <span class="error-msg" id="error"></span>
      </div-input>
    `;
  }

  readInput(e){
    if(e.target.value){
      console.log('evento change', e.target.value)
    }
  }

  requiredInput(e){
    let valueInput = e.target.value;
    let lengthInput = e.target.value.length;

    if(valueInput === ''){
      this.shadowRoot.getElementById("error").innerHTML = this.blankText;
    }else if(lengthInput < this.minLengthText){
      this.shadowRoot.getElementById("error").innerHTML = "Minimo " + this.minLengthText + " caracteres"
    }else if(lengthInput > this.maxLengthText){
      this.shadowRoot.getElementById("error").innerHTML = "Pasaste la longitud máxima de carácteres"
    }else{
      this.shadowRoot.getElementById("error").innerHTML = "";
    }
  }
}

customElements.define('input-lit', inputComponent);
