// button/src/button-lit.js
import { LitElement, html, css } from 'lit-element';

import { buttonStyle } from '../css/button-styles';
import { generalStyles } from '../css/general-styles';

export class ButtonComponent extends LitElement {

    static get styles(){
        return[
            buttonStyle,
            generalStyles
        ]
    }
    static get properties(){
        return{
            disabled: {type: Boolean}
        }
    }

    constructor(){
        super()
        this.disabled = false;
    }

  render() {
    return html`
    <button-default class="button-default" @click="${this.clickButton}" ?disabled=${this.disabled}>Default</button-default>
    <button-blue class="button-primary" @click="${this.clickButton}" ?disabled=${this.disabled}>Primary</button-blue>
    <button-warning class="button-warning" @click="${this.clickButton}" ?disabled=${this.disabled}>Warning</button-warning>
    <button-secondary class="button-secondary" @click="${this.clickButton}" ?disabled=${this.disabled}>Secondary</button-secondary>
    <button-disabled class="button-disabled" disabled>Disabled</button-disabled>
    `;
  }

  clickButton(){
      console.log('Diste click en el botón')
  }
}

customElements.define('button-lit', ButtonComponent);
