// select/src/select-lit.js
import { LitElement, html, css } from 'lit-element';

import { selectStyles } from '../css/select-styles';

export class selectComponent extends LitElement {

static get styles(){
  return[
    selectStyles
  ]
}

static get properties(){
  return{
    blanText: {type: String},
    disabled: {type: Boolean},
    items: {type: Array}
  }
}

constructor(){
  super()
  this.blanText = "Campo obligatorio";
  this.disabled = true;
  this.items = [
    {
    id: 1,
    text: 'Seleccione una opción',
    value: 'opcion'
   },
   {
    id: 2,
    text: 'Squirtle',
    value: 'agua'
   },
   {
    id: 3,
    text: 'Pikachu',
    value: 'electrico'
   },
   {
    id: 4,
    text: 'Charmander',
    value: 'fuego'
   }
]
}


  render() {
    return html`
      <div-select>
        <label class="required">Select general</label>
          <div class="space"></div>
          <select
          class="colorSelect"
          @input="${this.readSelect}"
          @change="${this.requiredSelecte}">
            <option value="none">Select país</option>
            <option value="au">Australia</option>
            <option value="ca">Canada</option>
            <option value="usa">USA</option>
          </select>
          <div class="space"></div>
          <span class="error-msg" id="error"></span>
      </div-select>
      <br><br>
      <div-select>
      <label class="required">Select general</label>
        <div class="space"></div>
        <select
        class="colorSelect"
        @input="${this.readSelect}"
        @change="${this.requiredSelecte}"
        ?disabled="${this.disabled}">
          <option value="none">Select país</option>
          <option value="au">Australia</option>
          <option value="ca">Canada</option>
          <option value="usa">USA</option>
        </select>
        <div class="space"></div>
      <span class="error-msg" id="error"></span>
      <br><br>
      <div-select>
      <label class="required">Select general</label>
        <div class="space"></div>
        <select
        class="colorSelect"
        @input="${this.readSelect}"
        @change="${this.requiredSelecte}">
        ${this.items.map(pokemon => html`
          <option value="${pokemon.value}">${pokemon.text}</option>
        `)}
        </select>
        <div class="space"></div>
        <span class="error-msg" id="error"></span>
    </div-select>
    `;
  }

  readSelect(e){
    if(e.target.value){
      console.log("evento change", e.target.value)
    }
  }

  requiredSelecte(e){
    let valueSelect = e.target.value;

    if(valueSelect === 'none'){
      this.shadowRoot.getElementById("error").innerHTML = this.blanText;
    }else{
      this.shadowRoot.getElementById("error").innerHTML = "";
    }
  }
}

customElements.define('select-lit', selectComponent);
