// button/css/button-styles.js
import { css } from "lit-element";

export const buttonStyle = css `

.button-default {
    background-color: #f3efee;
    border: none;
    color: black;
    padding: 10px 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    border-radius: 5px;
    font-family: "Lucida Console", "Courier New", monospace;
  }
  
  .button-default:hover {
      box-shadow: 0 6px 8px 0 rgba(0,0,0,0.24);
  }
  
  button-default:disabled,
  button-default[disabled]{
    background-color: #cccccc;
    color: #f6f6f6;
    pointer-events:none;
  }
  
  .button-primary {
    background-color: #9EBE1D;
    border: none;
    color: white;
    padding: 10px 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    border-radius: 5px;
    font-family: "Lucida Console", "Courier New", monospace;
  }
  
  .button-primary:hover {
    box-shadow: 0 6px 8px 0 rgba(0,0,0,0.24);
  }
  
  button-blue:disabled,
  button-blue[disabled]{
    background-color: #cccccc;
    color: #f6f6f6;
    pointer-events:none;
  }
  
  .button-warning{
    background-color: #f71906;
    border: none;
    color: white;
    padding: 10px 10px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 14px;
    border-radius: 5px;
    font-family: "Lucida Console", "Courier New", monospace;
  }
  
  .button-warning:hover {
    box-shadow: 0 6px 8px 0 rgba(0,0,0,0.24);
    }
  
    button-warning:disabled,
    button-warning[disabled]{
      background-color: #cccccc;
      color: #f6f6f6;
      pointer-events:none;
    }
  
    .button-disabled{
      background-color: #f3efee;
      border: none;
      padding: 10px 10px;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      font-size: 14px;
      border-radius: 5px;
      font-family: "Lucida Console", "Courier New", monospace;
    }
  
    button-disabled:disabled,
    button-disabled[disabled]{
      background-color: #cccccc;
      color: #f6f6f6;
      pointer-events:none;
    }
  
    .button-secondary{
      background-color: #72d0fb;
      border: none;
      color: white;
      padding: 10px 10px;
      text-align: center;
      text-decoration: none;
      display: inline-block;
      font-size: 14px;
      border-radius: 5px;
      font-family: "Lucida Console", "Courier New", monospace;
    }
    
    .button-secondary:hover {
      box-shadow: 0 6px 8px 0 rgba(0,0,0,0.24);
      }
    
      button-secondary:disabled,
      button-secondary[disabled]{
        background-color: #cccccc;
        color: #f6f6f6;
        pointer-events:none;
      }
  
      button-default-circle {
        height: 60px;
        width: 60px;
        background-color: #f3efee;
        border-radius: 50%;
        display: inline-block;
    }
  
    button-default-circle:disabled,
    button-default-circle[disabled]{
      background-color: #cccccc;
      color: #f6f6f6;
      pointer-events:none;
    }
  
    button-blue-circle {
      height: 60px;
      width: 60px;
      background-color: #9EBE1D;
      border-radius: 50%;
      display: inline-block;
  }
  
  button-blue-circle:disabled,
  button-blue-circle[disabled]{
    background-color: #cccccc;
    color: #f6f6f6;
    pointer-events:none;
  }
  
  button-warning-circle {
    height: 60px;
    width: 60px;
    background-color: #f71906;
    border-radius: 50%;
    display: inline-block;
  }
  
  button-warning-circle:disabled,
  button-warning-circle[disabled]{
  background-color: #cccccc;
  color: #f6f6f6;
  pointer-events:none;
  }
  
  button-secondary-circle {
    height: 60px;
    width: 60px;
    background-color: #72d0fb;
    border-radius: 50%;
    display: inline-block;
  }
  
  button-secondary-circle:disabled,
  button-secondary-circle[disabled]{
  background-color: #cccccc;
  color: #f6f6f6;
  pointer-events:none;
  }
  
  button-disable-circle {
    height: 60px;
    width: 60px;
    background-color: #f3efee;
    border-radius: 50%;
    display: inline-block;
  }
  
  button-disable-circle:disabled,
  button-disable-circle[disabled]{
  background-color: #cccccc;
  color: #f6f6f6;
  pointer-events:none;
    
`;
